api/v1alpha1/apidefinition_webhook.go

```diff
@@ -202,14 +202,24 @@ func (in *ApiDefinition) ValidateDelete() error {
 
 func (in *ApiDefinition) validateTarget() field.ErrorList {
 	var all field.ErrorList
-	// we make sure targets are properly set. We can either have a normal url
-	// darget or a looping target or both.
-	if in.Spec.Proxy.TargetURL == "" && in.Spec.Proxy.TargeInternal == nil {
-		all = append(all,
-			field.Required(path("proxy", "target_url"),
-				"can't be emptry",
-			),
-		)
+	// TargetURL is only allowed to be an empty string when we have GraphQL
+	// API, or we are targeting internal API by its name and namespace.
+	if in.Spec.Proxy.TargetURL == "" {
+		if in.Spec.GraphQL != nil {
```

Comment:
We can merge both the checks together
```suggestion
		if in.Spec.GraphQL != nil && !in.Spec.GraphQL.Enabled {
```